src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+            return ImmutableArray<Assembly>.Empty;
+        }
+
+        private static IEnumerable<Type> GetLoadableTypes(Assembly assembly)
+        {
+            if (assembly == null) throw new ArgumentNullException(nameof(assembly));
+            try
+            {
+                return assembly.GetTypes();
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                Console.WriteLine($"Failed to load all types from {assembly.FullName}. Will proceed with types loaded successfully.");
+                return e.Types.Where(t => t != null);
+            }
+        }
+
+        private static ImmutableArray<DiagnosticAnalyzer> GetAllAnalyzers(IEnumerable<Assembly> assemblies)
+        {
+            var analyzers = ImmutableArray.CreateBuilder<DiagnosticAnalyzer>();
+            foreach (var diagnosticAnalyzer in GetAllSubclasses<DiagnosticAnalyzer>(assemblies))
+            {
+                analyzers.Add(diagnosticAnalyzer);
+            }
+
+            return analyzers.ToImmutable();
+        }
+
+        private static ImmutableDictionary<string, ImmutableList<CodeFixProvider>> GetAllCodeFixers(IEnumerable<Assembly> assemblies)
+        {
+            Dictionary<string, ImmutableList<CodeFixProvider>> providers = new Dictionary<string, ImmutableList<CodeFixProvider>>();
+            foreach (var codeFixProvider in GetAllSubclasses<CodeFixProvider>(assemblies))
+            {
+                foreach (var diagnosticId in codeFixProvider.FixableDiagnosticIds)
+                {
+                    providers.AddToInnerList(diagnosticId, codeFixProvider);
+                }
+            }
+
+            return providers.ToImmutableDictionary();
+        }
+
+        private static IEnumerable<T> GetAllSubclasses<T>(IEnumerable<Assembly> assemblies)
+        {
+            var returnType = typeof(T);
+            foreach (var assembly in assemblies)
+            {
+                foreach (var type in GetLoadableTypes(assembly))
+                {
+                    if (type.IsSubclassOf(returnType) && !type.IsAbstract)
+                    {
+                        yield return (T)Activator.CreateInstance(type);
+                    }
+                }
+            }
+        }
+
+        private static async Task<ImmutableDictionary<ProjectId, ImmutableArray<Diagnostic>>> GetAnalyzerDiagnosticsAsync(Solution solution, string solutionPath, ImmutableArray<DiagnosticAnalyzer> analyzers, Options options, TelemetryCollector telemetryCollector, CancellationToken cancellationToken)
+        {
+            List<KeyValuePair<ProjectId, Task<ImmutableArray<Diagnostic>>>> projectDiagnosticTasks = new List<KeyValuePair<ProjectId, Task<ImmutableArray<Diagnostic>>>>();
+
+            // Make sure we analyze the projects in parallel
+            foreach (var project in solution.Projects)
+            {
+                if (project.Language != LanguageNames.CSharp)
```

Comment:
I dont think we want this.

Reply:
💭  We don't, but this could easily be moved to a follow-up issue.

Reply:
[https://github.com/dotnet/roslyn/issues/23108](https://github.com/dotnet/roslyn/issues/23108)

---
In reply to: [149862921](https://github.com/dotnet/roslyn/pull/23087#discussion_r149862921) [](ancestors = 149862921)